### PR TITLE
Swift: Fix some FPs from the sensitive data library

### PR DESCRIPTION
**Pull request by geoffw0**
Add tests for and fix some FPs caused by the sensitive data library.  These patterns have been observed in real world projects at various times.

In addition to removing likely FPs, the regexp changes do introduce a few new possible results (e.g. matching `mob_num`, `home_no`, `fax_num`), but based on a large MRVA run I don't expect this to have a big impact.

I should probably run DCA on this PR to confirm it removes the three FP results that show up there.

### CODE REVIEW

File: swift/ql/lib/codeql/swift/security/CleartextLoggingExtensions.qll [hunk 1]
```diff
@@ -29,6 +29,18 @@ private class DefaultCleartextLoggingSink extends CleartextLoggingSink {
...
+ * An barrier for cleartext logging vulnerabilities.
+ *  - encryption; encrypted values are not cleartext.
+ *  - booleans; these are more likely to be settings, rather than actual sensitive data.
+ */
+private class CleartextLoggingDefaultBarrier extends CleartextLoggingBarrier {
+  CleartextLoggingDefaultBarrier() {
+    this.asExpr() instanceof EncryptedExpr or
+    this.asExpr().getType().getUnderlyingType() instanceof BoolType
+  }
+}
```
Comment (geoffw0) [inline swift/ql/lib/codeql/swift/security/CleartextLoggingExtensions.qll:42]: @mathiasvp do you happen to know whether a potentially large set of barriers (here: all expressions of `Bool` type) is likely to be a performance problem or will only be evaluated in the restricted context of nodes on a potential flow path?  (if you're not sure, I will spend a bit of time investigating this)

Comment (MathiasVP) [reply swift/ql/lib/codeql/swift/security/CleartextLoggingExtensions.qll:42]: We've done similar things in C++ in the past (i.e., block flow through all integral-typed expressions), and performance for this has always been fine. So I'm not concerned about this at all.

Comment (geoffw0) [reply swift/ql/lib/codeql/swift/security/CleartextLoggingExtensions.qll:42]: Awesome, thanks!

**Comment by geoffw0** [comment]
DCA shows two less alerts for `swift/cleartext-storage-database` on `signalapp__Signal-iOS`.  This is a good change, they are known false positives targeted by one of the changes in this PR.  Notably there were originally three false positives though, so I'd better investigate whether the third one still occurs or disappeared before this PR...

**Comment by geoffw0** [comment]
... by which I mean re-running a failed check ...

**Pull request merged**